File: 0009-palindrome-number/solution.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x <0:
            return False
        revnum = self.revnum(x)
        if revnum == x:
            return True
    def revnum(self,x):
        #example number = 121
        rev = 0
        while x != 0:
            rem = x % 10 # gives value 1 \\ gives value 2 \\ gives value 1
            rev = rev*10 + rem# gives value 1
            x = x // 10
        print(rev)
        return rev



        

File: 0009-palindrome-number/solution.cpp

class Solution {
public:
    bool isPalindrome(int x) {
        int dup =x;
        int rev = 0;
        while (dup!=0){
            int digit = dup%10;
            dup = dup/10;
            if (rev > INT_MAX/10 || rev < INT_MIN/10){
                return false;
            };
            rev = rev*10 + digit;
        }
        if (rev != x || rev <0){
            return false;
        }
        return true;
        
    }
};